ps_wirepayment.php

```diff
@@ -442,4 +443,21 @@ public function getTemplateVarInfos()
             'bankwireCustomText' => $bankwireCustomText,
         );
     }
+
+    public function hookActionGetExtraMailTemplateVars($params)
+    {
+        if ($params['template']!='bankwire') {
+            return;
+        }
+
+        if (array_key_exists('{bankwire_owner}', $params['template_vars'])) {
+            return;
+        }
+        
+        $params['extra_template_vars'] = array(
```

Comment:
Is it safe to override everything that's inside `$params['extra_template_vars']`? There is a chance that other modules add data to it as well, right?

Btw, instead of defining new array of template vars, can you use `$this->extra_mail_vars`?

Reply:
Here is code from Mail.php
`
// Get extra template_vars
            $extraTemplateVars = [];
            Hook::exec(
                'actionGetExtraMailTemplateVars',
                [
                    'template' => $template,
                    'template_vars' => $templateVars,
                    'extra_template_vars' => &$extraTemplateVars,
                    'id_lang' => (int) $idLang,
                ],
                null,
                true
            );
`
I believe that I should change extra_template_vars or assign a new array to it.
Something like that:
'
$params['extra_template_vars']['{bankwire_owner}']  = Configuration::get('BANK_WIRE_OWNER');
//and so on
'
Both ways will work.